File: src/com/matrixx/settings/MatrixxLabSettings.java
/*
 * Copyright (C) 2023 havocOS-Extended
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.matrixx.settings;

import android.app.Activity;
import android.app.settings.SettingsEnums;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.ContentResolver;
import android.content.Context;
import android.content.DialogInterface;
import android.content.res.Resources;
import android.content.Intent;
import android.content.pm.UserInfo;
import android.os.Bundle;
import android.os.UserManager;
import android.view.View;
import android.provider.SearchIndexableResource;
import android.provider.Settings;
import android.text.TextUtils;
import android.text.format.DateFormat;
import androidx.preference.*;

import com.android.internal.logging.nano.MetricsProto;
import com.android.settings.R;
import com.android.settings.Utils;
import com.android.settings.dashboard.DashboardFragment;
import com.android.settings.search.BaseSearchIndexProvider;
import com.android.settings.widget.EntityHeaderController;
import com.android.settingslib.core.AbstractPreferenceController;
import com.android.settingslib.core.lifecycle.Lifecycle;
import com.android.settingslib.search.SearchIndexable;
import com.android.settingslib.widget.LayoutPreference;

import com.matrixx.settings.MatrixxLabController;
import com.android.settings.SettingsPreferenceFragment;
import com.android.settingslib.search.Indexable;

import java.time.format.DateTimeFormatter;
import java.time.LocalTime;
import java.util.Arrays;
import java.util.ArrayList;
import java.util.Collection;
import java.util.HashSet;
import java.util.List;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class MatrixxLabSettings extends DashboardFragment {

        private static final String LOG_TAG = "MatrixxLabSettings";

    @Override
    protected int getPreferenceScreenResId() {
        return R.xml.matrixx_lab_settings;
    }

    @Override
    public int getMetricsCategory() {
        return MetricsProto.MetricsEvent.MATRIXX;
    }

    @Override
    public int getHelpResource() {
        return R.string.help_uri_about;
    }
    
    @Override
    public void onStart() {
        super.onStart();
    }
    
    @Override
    protected String getLogTag() {
        return LOG_TAG;
    }
    
    @Override
    protected List<AbstractPreferenceController> createPreferenceControllers(Context context) {
        return buildPreferenceControllers(context, this /* fragment */, getSettingsLifecycle());
    }
    
    private static List<AbstractPreferenceController> buildPreferenceControllers(
            Context context, MatrixxLabSettings fragment, Lifecycle lifecycle) {
        final List<AbstractPreferenceController> controllers = new ArrayList<>();
        controllers.add(new MatrixxLabController(context));
        return controllers;
    }

    /**
     * For Search.
     */
    public static final BaseSearchIndexProvider SEARCH_INDEX_DATA_PROVIDER =
            new BaseSearchIndexProvider(R.xml.matrixx_lab_settings) {
                @Override
                public List<AbstractPreferenceController> createPreferenceControllers(
                        Context context) {
                    return buildPreferenceControllers(context, null /* fragment */,
                            null /* lifecycle */);
                }
            };
 } 

File: src/com/matrixx/settings/MatrixxLabSettings.kt
/*
 * Copyright (C) 2023 havocOS-Extended
 * Copyright (C) 2026 ProjectMatrixx
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.matrixx.settings

import android.content.Context
import com.android.internal.logging.nano.MetricsProto
import com.android.settings.R
import com.android.settings.dashboard.DashboardFragment
import com.android.settings.search.BaseSearchIndexProvider
import com.android.settingslib.core.AbstractPreferenceController
import com.android.settingslib.core.lifecycle.Lifecycle

class MatrixxLabSettings : DashboardFragment() {

    companion object {
        const val CATEGORY_MATRIXX = "com.android.settings.category.ia.matrixx"
        private const val LOG_TAG = "MatrixxLabSettings"

        /**
         * For Search.
         */
        @JvmField
        val SEARCH_INDEX_DATA_PROVIDER =
            object : BaseSearchIndexProvider(R.xml.matrixx_lab_settings) {
                override fun createPreferenceControllers(
                    context: Context
                ): List<AbstractPreferenceController> {
                    return buildPreferenceControllers(context, null, null)
                }
            }

        private fun buildPreferenceControllers(
            context: Context,
            fragment: MatrixxLabSettings?,
            lifecycle: Lifecycle?
        ): List<AbstractPreferenceController> {
            val controllers = ArrayList<AbstractPreferenceController>()

            // Keep your controller intact
            controllers.add(MatrixxLabController(context))

            return controllers
        }
    }

    override fun getPreferenceScreenResId(): Int {
        return R.xml.matrixx_lab_settings
    }

    override fun getMetricsCategory(): Int {
        return MetricsProto.MetricsEvent.MATRIXX
    }

    override fun getHelpResource(): Int {
        return R.string.help_uri_about
    }

    override fun onStart() {
        super.onStart()
    }

    override fun getLogTag(): String {
        return LOG_TAG
    }

    override fun createPreferenceControllers(
        context: Context
    ): List<AbstractPreferenceController> {
        return buildPreferenceControllers(context, this, settingsLifecycle)
    }
}